state/model.go

```diff
@@ -354,6 +354,15 @@ func (st *State) NewModel(args ModelArgs) (_ *Model, _ *State, err error) {
 		assertCloudCredentialOp,
 	}
 	ops := append(prereqOps, modelOps...)
+
+	if cloudCredentialOk {
```

Comment:
`if arg.CloudCredential != (names.CloudCredentialTag{}) {`?

"cloudCredentialOk" doesn't really convey to me that "a cloud credential was supplied"; it suggests more that it's valid, but an invalid cloud credential should already have been rejected.